tests/unit_tests/libs/controllers/test_predictor_timeseries.py

```diff
@@ -318,3 +318,67 @@ def test_split_models(self):
             assert str(row['order_ai_id']) == str(columns_test[2][i])
             assert str(row['4_valued_group_by']) == str(columns_test[3][i])
             assert str(row['5_valued_group_by']) == str(columns_test[4][i])
+
+    def test_infer(self):
+        ts_hours = 12
+        data_len = 120
+        train_file_name = os.path.join(self.tmp_dir, 'train_data.csv')
+        test_file_name = os.path.join(self.tmp_dir, 'test_data.csv')
+
+        features = generate_value_cols(['date', 'int', 'int', 'true'], data_len, ts_hours * 3600)
+        features[-1][0] = 'make_predictions'  # add make_predictions column as mindsdb would
+        labels = [generate_timeseries_labels(features[:-1])]
+
+        feature_headers = list(map(lambda col: col[0], features))
+        label_headers = list(map(lambda col: col[0], labels))
+
+        # Create the training dataset and save it to a file
+        columns_train = list(map(lambda col: col[1:int(len(col) * 3 / 4)], features))
```

Comment:
Is `make_predictions` column required for train data? If `yes` I don't think that it will be good for users. As I understand that `make_predictions` column needs for TS prediction only, not for training.

Reply:
That's a good question. It is not needed in train data. Notice in this test the last column of the dataset (which is "useless" as it contains True values only) is replaced in the `test_file_name` case by a `make_predictions = False` column, which complies with the behavior you expect: only in `predict()` call do we need `make_predictions`.